### PR TITLE
[next] Auto-store prompted for values in team config

### PR DESCRIPTION
**Pull request by t1m0thyj**
✅ ~~Please merge #687 first~~

Implements zowe/zowe-cli#923. In addition to adding the `ConfigAutoStore` class:
* Refactored `ConnectionPropsForSessCfg.addPropsOrPrompt` to remove hardcoded list of prompt properties
* Added `parseString` option to `Config.save` to prevent undesired coercion of string values to other types
* Created utility methods `coercePropValue` and `getActiveProfileName` to reduce code duplication
* Fixed config-related Jest tests which mysteriously started failing by adding `fs` mock

Thanks to @zFernand0 for writing the integration tests and @gejohnston for suggesting the `fs` mock 🙂 

I have reviewed the SonarCloud scan results. The security hotspot is a false positive. I have fixed some code smells and the remaining ones are unimportant IMO.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "lint": "eslint \"packages/**/*.ts\" \"**/__tests__/**/*.ts\"",
     "lint:packages": "eslint \"packages/**/*.ts\" --ignore-pattern \"**/__tests__/**/*.ts\"",
     "lint:tests": "eslint \"**/__tests__/**/*.ts\"",
+    "circularDependencyCheck": "madge -c lib",
```
Comment (zFernand0) [inline package.json:44]: I know this is probably already part of the gulp tasks, but curious if we will ever move things to just npm scripts and include this as part of the build 😋

File: packages/config/src/api/ConfigLayers.ts [hunk 2]
```diff
@@ -187,6 +187,10 @@ export class ConfigLayers extends ConfigApi {
             }
         }
 
+        if (cnfg.autoStore) {
+            layer.properties.autoStore = true;
+        }
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply packages/config/src/api/ConfigLayers.ts:192]: I think it's ok to give priority to autoStore=true 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: package.json (hunk 1 above)
Comment (awharn) [reply package.json:44]: I hope things get moved eventually. It'd be nice to be able to remove gulp.

Comment (t1m0thyj) [reply package.json:44]: Based on internal discussion, seems like we have sequential tasks using Gulp that don't need to be, and their performance could be improved by degulpifying them 🙂 

For more complex tasks like parallel commands (watch = tsc + lint) or web help bundle generation, it's probably fine to keep using Gulp for now.

Comment (t1m0thyj) [reply package.json:44]: Created #686 to track this

**Review by awharn — APPROVED** [review]

File: packages/config/src/Config.ts [hunk 3]
```diff
@@ -385,9 +386,13 @@ export class Config {
...
      * @param value Property value
-     * @param opts Include `secure: true` to store the property securely
+     * @param opts Optional parameters to change behavior
+     * * `parseString` - If true, strings will be converted to a more specific
+     *                   type like boolean or number when possible
+     * * `secure` - If true, the property will be stored securely.
+     *              If false, the property will be stored in plain text.
      */
-    public set(propertyPath: string, value: any, opts?: { secure?: boolean }) {
+    public set(propertyPath: string, value: any, opts?: { parseString?: boolean; secure?: boolean }) {
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/config/src/Config.ts:395]: Yes, thanks for calling it out 🙂 Discussed with @crawr and this change should be ok for Zowe Explorer. I've also added it as a next breaking change in the changelog.

**Review by gejohnston — APPROVED** [review]

**Pull request merged**